src/widget/about/aboutfriendform.cpp

```diff
@@ -22,9 +25,13 @@ AboutFriendForm::AboutFriendForm(std::unique_ptr<IAboutFriend> _about, QWidget*
     connect(ui->removeHistory, &QPushButton::clicked, this, &AboutFriendForm::onRemoveHistoryClicked);
     about->connectTo_autoAcceptDirChanged([=](const QString& dir){ onAutoAcceptDirChanged(dir); });
 
+    connect(this, &AboutFriendForm::clearFriendChatLogArea, Widget::getInstance(), &Widget::onFriendChatAreaCleared);
```

Comment:
Please, think a bit in concept of isolation. Suppose every form can be started as separated application. Does AboutFriend form really need to know about main widget? Probably it shouldn't. How we can solve this issue? We can try to forward signal through creators of `AboutFriendForm` until `Widget` (another question is: should `Widget` know about history cleaning? But my answer is: current architecture too complex to avoid this, so let's allow `Widget` to know about it)
`AboutFriendForm` created in `FriendWidget`, who use model `FriendChatroom`. I think adding signal about chat history cleaning in model is fine (or even good). So, what is the final suggestion?

`FriendWidget` connect `AboutFriendForm::clearFriendChatLogArea` to `FriendChatroom::historyCleared` signal. `Widget` creates `FriendWidget` and `FriendChatroom` and here it can be connected to chatroom signal.

I hope this was useful :)